File: www.admin.com/frontend-app/src/app/model/backend/calendar/reservation.ts
import {ApplicationUser} from "../auth/application-user";

export class Reservation {

  public id: number;
  public user: ApplicationUser;
  public event: string;
  public date: Date;
  public childName: string;

  constructor(date: Date) {
    this.date = date;
  }
}

File: www.admin.com/core-api/src/main/java/com/base/coreapi/model/calendar/Reservation.java
package com.base.coreapi.model.calendar;

import com.base.coreapi.model.auth.ApplicationUser;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;

@Data
@Entity
@NoArgsConstructor
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @ManyToOne
    private ApplicationUser user;

    private String event;

    private Date date;
}
